com.arc.embeddedcdt/src/com/arc/embeddedcdt/gui/DebuggerGroupContainer.java

```diff
@@ -305,6 +323,115 @@ public void propertyChange(PropertyChangeEvent event) {
     });
   }
 
+  private void updateFtdiCoreCombo() {
+    getFtdiCoreCombo().removeAll();
+    java.util.List<FtdiCore> cores = getFtdiDevice().getCores();
+    String text = cores.get(0).toString();
+    for (FtdiCore core : cores) {
+        getFtdiCoreCombo().add(core.toString());
+        if (getFtdiCore() == core) {
+            /*
+             * Should select current ftdiCore if it is present in cores list in order to be able
+             * to initialize from configuration. Otherwise ftdiCore field will be rewritten to
+             * the selected core when we initialize FTDI_DeviceCombo
+             */
+            text = core.toString();
+        }
+    }
+    getFtdiCoreCombo().setText(text);
+  }
+
+  private void createTabItemCom(final Composite compositeCom) {
+    Label label = new Label(compositeCom, SWT.LEFT);
+    label.setText("Development system:");
+    setFtdiDeviceCombo(new Combo(compositeCom, SWT.None | SWT.READ_ONLY));// 1-2 and 1-3
+
+    GridData gridDataJtag = new GridData(GridData.BEGINNING);
+    gridDataJtag.widthHint = 220;
+    gridDataJtag.horizontalSpan = 2;
+    getFtdiDeviceCombo().setLayoutData(gridDataJtag);
+
+    for (FtdiDevice i : FtdiDevice.values())
+        getFtdiDeviceCombo().add(i.toString());
+    getFtdiDeviceCombo().setText(getFtdiDevice().toString());
+
+    getFtdiDeviceCombo().addModifyListener(new ModifyListener() {
+        public void modifyText(ModifyEvent event) {
```

Comment:
Is it possible to extract those two `modifyText` functions, so they will not be inner functions.